Search functions by common name, exact common name, by extension, by …

ovandononi opened the pull request:
Changes Search functions where it is possible to search files and directories by a specific criteria

ovandononi committed:
Search functions by common name, exact common name, by extension, by size (MB)

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+#import File
+import os
+
+class Asset (object):
```
Please add a new line before of the Class definition line, it should contains 2 lines before of it

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+    def get_is_file(self):
+        return self._is_file
+
+class File(Asset):
```
Please add a new line before of the Class definition line, it should contains 2 lines before of it

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+    def get_extension(self):
+        return self._extension
+
+class Directory(Asset):
```
Please add a new line before of the Class definition line, it should contains 2 lines before of it

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+#import File
```
Remove this comment please

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+
+    def get_file_name(self):
+        return self._file_name
+    def get_extension(self):
```
Please add a new line before of the method definition line, it should contains 1 line before of it

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+
+class Directory(Asset):
+
+    def __init__(self, path, dir_name,is_file):
```
Please add a space after the comma, just like below
def __init__(self, path, dir_name,is_file):

Miquilin commented on src/com/jalasoft/search_files/search/asset.py:
```diff
@@ -0,0 +1,40 @@
+
+    def __init__(self, path, dir_name,is_file):
+        super().__init__(path, is_file)
+        self.count_content=0
```
Assignation "=" should contain a space before and after of it, just like:
self.count_content = 0
self.dir_name = dir_name
Please change it.

Miquilin commented on src/com/jalasoft/search_files/search/factory_asset.py:
```diff
@@ -0,0 +1,25 @@
+#import File
```
Please remove this comment

Miquilin commented on src/com/jalasoft/search_files/search/factory_asset.py:
```diff
@@ -0,0 +1,25 @@
+#import File
+import os
+from src.com.jalasoft.search_files.search.asset import Asset, File, Directory
```
"Asset" is not being used, maybe it could be removed.

Miquilin commented on src/com/jalasoft/search_files/search/factory_asset.py:
```diff
@@ -0,0 +1,25 @@
+        self._list_actual_directories_and_files = []
+
+    def create_asset(self, path):
+        result = []
```
Apparenlty the "result" variable is not being used, maybe we can remove it

Miquilin commented on src/com/jalasoft/search_files/search/factory_asset.py:
```diff
@@ -0,0 +1,25 @@
+        result = []
+        for root, directories, files in os.walk(path):
+            for dir in directories:
+                directory = Directory(os.path.join(root, dir),dir, False)
```
please add a space after the comma, just like below
directory = Directory(os.path.join(root, dir), dir, False)

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+import os
```
this library is not being used, maybe we can remove it

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+import os
+from pathlib import Path
```
This library is not being used, maybe we can remove it

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+import os
+from pathlib import Path
+from src.com.jalasoft.search_files.search.factory_asset import FactoryAsset
+from src.com.jalasoft.search_files.search.asset import Asset, File, Directory
```
"Asset" is not being used, maybe we can remove it

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+    def __init__(self):
+        self._path = ""
+
+
```
Please remove this extra line

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+                Array [Asset]: The return directories and/or files list.
+
+                """
+        result= []
```
Please add a space before "=", just like below
result = []

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+        factory_asset = FactoryAsset()
+        factory_asset.create_asset(path_absolute)
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if which_search ==0:
```
add a space afer the "==", just like below:
if which_search == 0:

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+                Array [Asset]: The return directories and/or files list that have exact common name
+
+                """
+        result= []
```
please add a space before "=" just like:
result = []

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+        factory_asset.create_asset(path_absolute)
+        file_name, separator, extension = common_name.partition(".")
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if which_search ==0:
```
please add a space after the "==", just like:
if which_search == 0:

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if which_search ==0:
+                if isinstance(item, File):
+                    if extension != "" :
```
Please remove extra space, it should be like below:
if extension != "":

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+                Array [Asset]: The return directories and/or files list that have exact common extension
+
+                """
+        result= []
```
please add a space before of "=", just like below:
result = []

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+
+        return result
+
+    def search_files_by_less_than_a_size(self, path_absolute, size_on_megabites):
```
change the name of the variable from "size_on_megabites" to "size_on_megabytes"

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+        """
+                Args:
+                path_absolute (str): The first parameter.
+                size_on_megabites (int): The second parameter.
```
change the name of the variable from "size_on_megabites" to "size_on_megabytes"

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+        factory_asset.create_asset(path_absolute)
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if isinstance(item, File):
+                if  item.get_size()/(1024*1224) < size_on_megabites:
```
Please remove extra space after of "if", just like below:
if item.get_size()/(1024*1224) < size_on_megabites:

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+                Array [Asset]: The return directories and/or files list that have less than a size
+
+                """
+        result= []
```
Please add a space before of "=", jus like below
result = []

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if isinstance(item, File):
+
+                if  item.get_size()/(1024*1224) > size_on_megabites:
```
please remove the extra space just after the "if", like below
if item.get_size()/(1024*1224) < size_on_megabites:

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+                Array [Asset]: The return directories and/or files list that have bigger than a size
+
+                """
+        result= []
```
please add a space before of "="

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+                if  item.get_size()/(1024*1224) > size_on_megabites:
+                    result.append(item)
+                    print("item.get_size()/(1024*1224)", item.get_size() / (1024 * 1224))
+
```
pleas remove the extra lines

Miquilin commented on src/com/jalasoft/search_files/search/search.py:
```diff
@@ -0,0 +1,180 @@
+
+    def print_list_all(self, list_to_be_print):
+        for item in list_to_be_print:
+            print(item.get_path() , "test sixe" ,item.get_size()/(1024*1224) )
```
please change the position of the comma, just like below:
print(item.get_path() , "test sixe", item.get_size()/(1024*1224) )

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+import unittest
```
Please review the tabulation for this file

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+       files = ['file1.txt' , 'file2.dat' , 'file3.txt' , 'dir_file.txt' , 'file1_1.txt' , 'diledir2.txt' ,'none'  ]
+       # Count 11 files and directories
+       cont = len(directories) + len(files)
+       result_cont=len(search_test)
```
please add a space before and after the "="
result_cont = len(search_test)

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+       # Count 11 files and directories
+       cont = len(directories) + len(files)
+       result_cont=len(search_test)
+       self.assertEqual(cont,result_cont )
```
please add a space after the comma
self.assertEqual(cont, result_cont )

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+
+       # Count 6 files and directories
+       cont = len(directories) + len(files)
+       result_count=len(search_test)
```
please add a space before and after of "="
 result_count = len(search_test)

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+       # Count 6 files and directories
+       cont = len(directories) + len(files)
+       result_count=len(search_test)
+       self.assertEqual(cont,result_count )
```
Add the space after of comma
self.assertEqual(cont, result_count )

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+       files = ['file3']
+
+       cont = len(directories) + len(files)
+       result_count=len(search_test)
```
please add a space before and after the "="
result_count = len(search_test)

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+               for dir in directories:
+                   self.assertTrue(dir in item.get_dir_name())
+
+
```
please remove extra line

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+                    if dir in item.get_dir_name():
+                        self.assertTrue(dir in item.get_dir_name())
+
+
```
please remove extra lines

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+                    if file in item.get_file_name():
+                        self.assertTrue(file in item.get_file_name())
+
+
```
please remove extra line

Miquilin commented on test/search_test.py:
```diff
@@ -0,0 +1,182 @@
+                    if file in item.get_file_name():
+                        self.assertTrue(file in item.get_file_name())
+
+
```
please remove extra lines

ovandononi committed:
Changes related to code review observation #1

ovandononi committed:
Changes related to code review observation #1

remove unnecesary files

mtancara committed:
Fixing minor problems

I changed the value used to calculate the size to MB from 1224 to 1024
Fixed  pair of unitests and I removed a unused code on unitest class
